opentracing-spring-cloud/src/test/java/io/opentracing/contrib/spring/cloud/hystrix/HystrixTraceCommandTest.java

```diff
@@ -0,0 +1,182 @@
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.stereotype.Service;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+
+@SpringBootTest(classes = {HystrixTraceCommandTest.TestConfig.class})
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class HystrixTraceCommandTest {
+
+    @EnableAspectJAutoProxy(proxyTargetClass = true)
+    @EnableHystrix
+    @Configuration
+    //This is required for circuit breaker tests to work correctly
+    @EnableAutoConfiguration(exclude = {LoadBalancerAutoConfiguration.class, JmxAutoConfiguration.class})
+    static class TestConfig {
+        @Bean
+        public MockTracer mockTracer() {
+            return new MockTracer(new ThreadLocalActiveSpanSource(), MockTracer.Propagator.TEXT_MAP);
+        }
+
+        @Bean
+        public GreetingService greetingService() {
+            return new GreetingService();
+        }
+    }
+
+    @Service
+    static class GreetingService {
+
+        @Autowired
+        MockTracer mockTracer;
+
+        @HystrixCommand
+        public String sayHello() {
+            Span span = mockTracer.buildSpan("sayHello")
+                    .startManual();
+
+            span.finish();
+            return "Hello!!!";
+        }
+
+        @HystrixCommand(fallbackMethod = "defaultGreeting")
+        public String sayHelloToUser(String user) {
+            Span span = mockTracer.buildSpan("sayHelloToUser")
+                    .startManual();
+            try {
+                throw new IllegalStateException("Thrown Purposefully");
+            } finally {
+                span.finish();
+            }
+        }
+
+        public String defaultGreeting(String user) {
+            Span span = mockTracer.buildSpan("sayHelloToUser")
+                    .withTag("myTag", "fallback")
+                    .startManual();
+            try {
+                return "Hi(fallback)";
+            } finally {
+                span.finish();
+            }
+        }
+    }
+
+    @Autowired
+    private MockTracer mockTracer;
+
+    @Autowired
+    private GreetingService greetingService;
+
+    @BeforeClass
+    @AfterClass
+    public static void reset() {
+        HystrixPlugins.reset();
+    }
+
+    @Before
+    public void before() {
+        mockTracer.reset();
+    }
+
+    @Test
+    public void test_without_circuit_breaker() throws Exception {
+
+        try (ActiveSpan activeSpan = mockTracer.buildSpan("test_without_circuit_breaker")
+                .startActive()) {
+            String response = greetingService.sayHello();
+            assertThat(response).isNotNull();
```

Comment:
move the asserts out of `try` block and add asserts on hystrix tags(applies for other tests too)

Reply:
I cant move out side try-with-resources, as am not able to get the finished spans and all test cases fail

Reply:
tags can be tested only for `HystrixTrace` command, for others its just hooked with TraceCallable - so no tags will there